tests/org/intellij/erlang/completion/ErlangCompletionTest.java

```diff
@@ -185,6 +185,18 @@ public boolean value(String s) {
   public void testBehaviour()         { doCheckResult("-behaviou<caret>", "-behaviour(<caret>)."); }
   public void testBehavior()          { doCheckResult("-behavior<caret>", "-behavior(<caret>)."); }
 
+  public void testAppFile() {
+    myFixture.configureByText("a.app", "{ap<caret>}.");
+    myFixture.completeBasic();
+    myFixture.checkResult("{application}.");
+  }
+
+  public void testAppFileParameter() {
+    myFixture.configureByText("a.app", "{application, name, [{re<caret>}]}.");
+    myFixture.completeBasic();
+    myFixture.checkResult("{application, name, [{registered<caret>}]}.");
+  }
+
```

Comment:
These tests do not cover false-positives - make sure you don't provide your completion variants in unsuitable places.